File: stored-procedure/dotnet/Controllers/UserController.cs
using Microsoft.AspNetCore.Mvc;
using CompanyApp.UserManagement.Services;

namespace CompanyApp.UserManagement.Controllers
{
    [ApiController]
    [Route("api")]
    public class UserController : ControllerBase
    {
        private readonly UserManagementService _userService;

        public UserController(UserManagementService userService)
        {
            _userService = userService;
        }

        [HttpPost("users/search")]
        public async Task<IActionResult> SearchUsers([FromForm] string searchTerm)
        {
            try
            {
                var results = await _userService.SearchUsersAsync(searchTerm);
                return Ok(results);
            }
            catch (Exception ex)
            {
                return BadRequest(new { error = ex.Message });
            }
        }

        [HttpPost("orders/customer")]
        public async Task<IActionResult> GetCustomerOrders([FromForm] string customerId)
        {
            try
            {
                var results = await _userService.GetUserOrdersAsync(customerId);
                return Ok(results);
            }
            catch (Exception ex)
            {
                return BadRequest(new { error = ex.Message });
            }
        }

        [HttpPost("access/validate")]
        public async Task<IActionResult> ValidateAccess([FromForm] string username, [FromForm] string accessLevel)
        {
            try
            {
                var result = await _userService.ValidateUserAccessAsync(username, accessLevel);
                return Ok(result);
            }
            catch (Exception ex)
            {
                return BadRequest(new { error = ex.Message });
            }
        }
        [HttpPost("users/search-secure")]
        public async Task<IActionResult> SearchUsersSecure([FromForm] string searchTerm)
        {
            try
            {
                var results = await _userService.SearchUsersSecureAsync(searchTerm);
                return Ok(results);
            }
            catch (Exception)
            {
                return BadRequest(new { error = "An error occurred while processing your request" });
            }
        }

        [HttpPost("orders/customer-secure")]
        public async Task<IActionResult> GetCustomerOrdersSecure([FromForm] string customerId)
        {
            try
            {
                var results = await _userService.GetUserOrdersSecureAsync(customerId);
                return Ok(results);
            }
            catch (Exception)
            {
                return BadRequest(new { error = "An error occurred while processing your request" });
            }
        }
    }
}

File: stored-procedure/spring-boot/src/main/java/com/sqlilab/storedprocedure/controller/UserController.java
package com.sqlilab.storedprocedure.controller;

import com.sqlilab.storedprocedure.service.UserManagementService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;

import java.util.List;
import java.util.Map;

@Controller
public class UserController {

    @Autowired
    private UserManagementService userService;

    @GetMapping("/")
    public String index(Model model) {
        return "index";
    }

    @GetMapping("/users")
    public String userSearch(Model model) {
        model.addAttribute("title", "User Search");
        model.addAttribute("description", "Search for users in the system by username or email.");
        return "users";
    }

    @PostMapping("/api/users/search")
    @ResponseBody
    public List<Map<String, Object>> searchUsers(@RequestParam String searchTerm) {
        return userService.searchUsers(searchTerm);
    }

    @GetMapping("/orders")
    public String orderManagement(Model model) {
        model.addAttribute("title", "Order Management");
        model.addAttribute("description", "View customer orders and order history.");
        return "orders";
    }

    @PostMapping("/api/orders/customer")
    @ResponseBody
    public List<Map<String, Object>> getCustomerOrders(@RequestParam String customerId) {
        return userService.getUserOrders(customerId);
    }

    @GetMapping("/access")
    public String accessControl(Model model) {
        model.addAttribute("title", "Access Validation");
        model.addAttribute("description", "Validate user access permissions and roles.");
        return "access";
    }

    @PostMapping("/api/access/validate")
    @ResponseBody
    public Map<String, Object> validateAccess(@RequestParam String username, @RequestParam String accessLevel) {
        return userService.validateUserAccess(username, accessLevel);
    }

    @GetMapping("/secure")
    public String secureExamples(Model model) {
        model.addAttribute("title", "Secure Implementation Examples");
        return "secure";
    }

    @PostMapping("/api/users/search-secure")
    @ResponseBody
    public List<Map<String, Object>> searchUsersSecure(@RequestParam String searchTerm) {
        return userService.searchUsersSecure(searchTerm);
    }

    @PostMapping("/api/orders/customer-secure")
    @ResponseBody
    public List<Map<String, Object>> getCustomerOrdersSecure(@RequestParam String customerId) {
        return userService.getUserOrdersSecure(customerId);
    }
}
